src/bindgen/ir/constant.rs

```diff
@@ -151,7 +151,25 @@ impl Literal {
             syn::Expr::Lit(syn::ExprLit {
                 lit: syn::Lit::Char(ref value),
                 ..
-            }) => Ok(Literal::Expr(format!("'{}'", value.value()))),
+            }) => {
+                let v = value.value();
+                if v.len_utf8() == 1 {
+                    Ok(Literal::Expr(format!(
```

Comment:
Note that you can extract common parts out:

```rust
Ok(Literal::Expr(match v as u32 {
  0..=255 => format!("'{}'", v.escape_default()),
  other_code => format!(r"L'\u{:X}'", other_code),
}))
```

Reply:
nice!